--- ../../../../Applications/util/ps.c	2024-02-12 16:06:26.445511000 -0700
+++ ps.c	2024-03-31 10:58:57.708871714 -0600
@@ -35,6 +35,7 @@
 #include <pwd.h>
 #include <fcntl.h>
 #include <time.h>
+#include "progbase.h"
 
 #define TTY_MAJOR	2		/* Be nice if we didn't have to
 					   just know this */
@@ -66,6 +67,9 @@
 #define OF_TIME		8192		/* Cumulative execution time */
 #define OF_CMD		16384		/* Command line */
 
+#define BLOCKSIZE 4096
+#define alignup(v,a) (uint8_t*)((intptr_t)((v) + (a)-1) & ~((a)-1))
+
 static const char *month = "JanFebMarAprMayJunJulAugSepOctNovDec";
 
 static char mapstat(char s)
@@ -348,7 +352,11 @@
 	if (outflags & OF_ADDR)
 		fputs("    - ", stdout);
 	if (outflags & OF_SZ)
-		fputs("    - ", stdout);
+	{
+		long size = pp->p_top - PROGBASE;
+		size = (long)alignup(size, BLOCKSIZE) / 1024;
+		printf("%5d ", size);
+	}
 	if (outflags & OF_WCHAN) {
 		if (pp->p_status > 2)
 			printf(" %4x ", (unsigned int)pp->p_wait);
