--- cromemco-fdc.c~	2019-08-25 21:28:13.000000000 +0100
+++ cromemco-fdc.c	2019-08-25 21:32:17.797997501 +0100
@@ -82,10 +82,10 @@
 
 /* these are our disk drives, 8" SS SD initially */
 static Diskdef disks[4] = {
-	{ "drivea.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE },
-	{ "driveb.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE },
-	{ "drivec.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE },
-	{ "drived.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE }
+	{ "drivea.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE, SPT8SD },
+	{ "driveb.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE, SPT8SD },
+	{ "drivec.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE, SPT8SD },
+	{ "drived.dsk", LARGE, SINGLE, ONE, TRK8, SPT8SD, SPT8SD, READWRITE, SPT8SD }
 };
 
 /*
@@ -184,6 +184,7 @@
 		disks[disk].tracks = TRK8;
 		disks[disk].sectors = SPT8DD;
 		disks[disk].sec0 = SPT8SD;
+		disks[disk].disk_d0 = SINGLE;
 		break;
 
 	case 1256704:		/* 8" DS DD */
@@ -193,10 +194,21 @@
 		disks[disk].tracks = TRK8;
 		disks[disk].sectors = SPT8DD;
 		disks[disk].sec0 = SPT8SD;
+		disks[disk].disk_d0 = SINGLE;
+		break;
+
+	case 1261568:		/* 8" DS DD no SD track */
+		disks[disk].disk_t = LARGE;
+		disks[disk].disk_d = DOUBLE;
+		disks[disk].disk_s = TWO;
+		disks[disk].tracks = TRK8;
+		disks[disk].sectors = SPT8DD;
+		disks[disk].sec0 = SPT8SD;
+		disks[disk].disk_d0 = DOUBLE;
 		break;
 
 	default:
-		//printf("disk image %s has unknow format\r\n", disks[disk].fn);
+		//printf("disk image %s has unknown format\r\n", disks[disk].fn);
 		disks[disk].disk_t = UNKNOWN;
 		break;
 	}
@@ -218,12 +230,18 @@
 
 	    /* double density */
 	    } else {
-		if (fdc_track == 0) {
-		  pos = (fdc_sec - 1) * SEC_SZSD;
+	    	if (disks[disk].disk_d0 == SINGLE) {
+			if (fdc_track == 0) {
+			  pos = (fdc_sec - 1) * SEC_SZSD;
+			  } else {
+			  pos = (disks[disk].sec0 * SEC_SZSD) +
+				((fdc_track - 1) * disks[disk].sectors * SEC_SZDD) +
+				((fdc_sec - 1) * SEC_SZDD);
+			}
 		} else {
-		  pos = (disks[disk].sec0 * SEC_SZSD) +
-			((fdc_track - 1) * disks[disk].sectors * SEC_SZDD) +
-			((fdc_sec - 1) * SEC_SZDD);
+			/* No weird SD track */			
+			pos = (fdc_track * disks[disk].sectors * SEC_SZDD) + 
+				(fdc_sec - 1) * SEC_SZDD;
 		}
 	    }
 
@@ -242,13 +260,19 @@
 
 	    /* double density */
 	    } else {
-		if ((fdc_track == 0) && (side == 0)) {
-		    pos = (fdc_sec - 1) * SEC_SZSD;
-		    goto done;
-		}
-		if ((fdc_track == 0) && (side == 1)) {
-		    pos = disks[disk].sec0 * SEC_SZSD + (fdc_sec - 1) * SEC_SZDD;
-		    goto done;
+	    	if (disks[disk].disk_d0 == SINGLE) {
+			if ((fdc_track == 0) && (side == 0)) {
+			    pos = (fdc_sec - 1) * SEC_SZSD;
+			    goto done;
+			}
+			if ((fdc_track == 0) && (side == 1)) {
+			    pos = disks[disk].sec0 * SEC_SZSD + (fdc_sec - 1) * SEC_SZDD;
+			    goto done;
+			}
+			pos = disks[disk].sec0 * SEC_SZSD + disks[disk].sectors * SEC_SZDD;
+			pos += (fdc_track - 1) * 2 * disks[disk].sectors * SEC_SZDD;
+		} else {
+			pos = fdc_track * 2 * disks[disk].sectors * SEC_SZDD;
 		}
 		pos = disks[disk].sec0 * SEC_SZSD + disks[disk].sectors * SEC_SZDD;
 		pos += (fdc_track - 1) * 2 * disks[disk].sectors * SEC_SZDD;
--- cromemco-fdc.h~	2019-08-25 21:28:13.000000000 +0100
+++ cromemco-fdc.h	2019-08-25 21:32:43.158423540 +0100
@@ -47,6 +47,7 @@
 	int sectors;			/* # sectors on tracks > 0 side 0 */
 	int sec0;			/* # sectors on track 0, side 0 */
 	enum Disk_mode disk_m;		/* R/W or R/O mode */
+	enum Disk_density disk_d0;	/* Density of track 0 */
 } Diskdef;
 
 extern BYTE fdc_flags;
